packages/host/app/components/operator-mode.gts

```diff
@@ -0,0 +1,166 @@
+import Component from '@glimmer/component';
+import { Card, Format, CardInstanceType } from 'https://cardstack.com/base/card-api';
+import { tracked } from '@glimmer/tracking';
+import Preview from './preview';
+import { action } from '@ember/object';
+import { on } from '@ember/modifier';
+import { fn } from '@ember/helper';
+import { service } from '@ember/service';
+import type CardService from '../services/card-service';
+import type RouterService from '@ember/routing/router-service';
+import getValueFromWeakMap from '../helpers/get-value-from-weakmap';
+import { eq, not } from '@cardstack/boxel-ui/helpers/truth-helpers';
+import { svgJar } from '@cardstack/boxel-ui/helpers/svg-jar';
+import cn from '@cardstack/boxel-ui/helpers/cn';
+import { restartableTask } from 'ember-concurrency';
+import { TrackedWeakMap } from 'tracked-built-ins';
+import { Model as CardRouteModel } from '../routes/card';
+
+interface Signature {
+  Args: {
+    firstCardInStack: CardRouteModel;
+  };
+}
+
+export default class OperatorMode extends Component<Signature> {
+  @tracked stack: Card[] = [];
+  formats: WeakMap<Card, Format> = new TrackedWeakMap<Card, Format>();
+
+  //A variable to store value of card field
+  //before in edit mode.
+  cardFieldValues: WeakMap<Card, Map<string, any>> = new WeakMap<Card, Map<string, any>>();
+  @service declare cardService: CardService;
+  @service declare router: RouterService;
+
+  constructor(owner: unknown, args: any) {
+    super(owner, args);
+    this.stack = [this.args.firstCardInStack.card!];
+  }
+
+  @action async edit(card: Card) {
+    await this.saveCardFieldValues(card);
+    this.formats.set(card, 'edit');
+  }
+
+  @action getFormat(card: Card): Format | undefined {
+    return this.formats.get(card);
+  }
+
+  @action async close(card: Card) {
+    await this.rollbackCardFieldValue(card);
+    let index = this.stack.indexOf(card);
+    this.stack.splice(index);
+    this.stack = this.stack;
+
+    if (this.stack.length <= 0) {
+      this.router.transitionTo('card', this.args.firstCardInStack);
+    }
+  }
+
+  @action async cancel(card: Card) {
+    await this.rollbackCardFieldValue(card);
+    this.formats.set(card, 'isolated');
+  }
+
+  @action async save(card: Card) {
+    await this.saveCardFieldValues(card);
+    this.write.perform(card);
+  }
+
+  private write = restartableTask(async (card: Card) => {
+    let updatedCard = await this.cardService.saveModel(card);
+    this.formats.set(updatedCard, 'isolated');
+  });
+
+  private async saveCardFieldValues(card: Card) {
```

Comment:
i could see us wanting to reuse this in the future. keep an eye out in other PR's that you review going forward to see if someone else wants to reuse this functionality--we can extract it into a shared lib when that happens.